pep-0622.rst

```diff
@@ -21,103 +21,254 @@ Resolution:
 Abstract
 ========
 
-This PEP proposes adding pattern matching statements [1]_ to Python in
-order to create more expressive ways of handling structured
-heterogeneous data.  The authors take a holistic approach, providing
-both static and runtime specifications.
+This PEP proposes to add a pattern matching statement to Python,
+inspired by similar syntax found in Scala and many other languages.
+
+The pattern syntax builds on Python’s existing syntax for sequence
+unpacking (e.g., ``a, b = value``), but is wrapped in a ``match``
+statement which compares its subject to several different “shapes”
+until one is found that fits. In addition to specifying the shape of a
+sequence to be unpacked, patterns can also specify the shape to be a
+mapping with specific keys, an instance of a given class with specific
+attributes, a specific value, or a wildcard. Patterns can be composed
+in several ways.
+
+Syntactically, a ``match`` statement contains a *subject* expression
+and one or more ``case`` clauses, where each case clause specifies a
+pattern (the overall shape to be matched), an optional “guard” (a
+condition to be checked if the pattern matches), and a code block to
+be executed if the case clause is selected.
+
+The rest of the PEP motivates why we believe pattern matching makes a
+good addition to Python, explains our design choices, and contains a
+precise syntactic and runtime specification. We also give guidance for
+static type checkers (and one small addition to the ``typing`` module)
+and discuss the main objections and alternatives that have been
+brought up during extensive discussion of the proposal, both within
+the group of authors and in the python-dev community. Finally, we
+discuss some possible extensions that might be considered in the
+future, once the community has ample experience with the currently
+proposed syntax and semantics.
+
+
+Overview
+========
 
-:pep:`275` and :pep:`3103` previously proposed similar constructs, and
-were rejected.  Instead of targeting the optimization of
-``if ... elif ... else`` statements (as those PEPs did), this design
-focuses on generalizing sequence, mapping, and object destructuring.
-It uses syntactic features made possible by :pep:`617`, which
-introduced a more powerful method of parsing Python source code.
+Since patterns are a new syntactical category, with their own rules
+and exceptions, and since they mix input (given values) and output
```

Comment:
To reduce confusion with capital-E Exceptions:

```suggestion
Since patterns are a new syntactical category with their own rules
and limitations, and since they mix input (given values) and output
```

Reply:
"special cases" maybe?

Reply:
```suggestion
Since patterns are a new syntactical category, with their own rules
and special cases, and since they mix input (given values) and output
```